homeassistant/components/balboa/climate.py

```diff
@@ -0,0 +1,161 @@
+        }
+        self._ha_to_balboa_blower_map = {
+            value: key for key, value in self._balboa_to_ha_blower_map.items()
+        }
+        self._balboa_to_ha_heatmode_map = {
+            self._client.HEATMODE_READY: HVAC_MODE_HEAT,
+            self._client.HEATMODE_RNR: HVAC_MODE_AUTO,
+            self._client.HEATMODE_REST: HVAC_MODE_OFF,
+        }
+        self._ha_heatmode_to_balboa_map = {
+            value: key for key, value in self._balboa_to_ha_heatmode_map.items()
+        }
+        scale = self._client.get_tempscale()
+        self._attr_preset_modes = self._client.get_heatmode_stringlist()
+        self._attr_supported_features = SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE
+        if self._client.have_blower():
+            self._attr_supported_features |= SUPPORT_FAN_MODE
+        self._attr_min_temp = self._client.tmin[self._client.TEMPRANGE_LOW][scale]
+        self._attr_max_temp = self._client.tmax[self._client.TEMPRANGE_HIGH][scale]
+        self._attr_temperature_unit = TEMP_FAHRENHEIT
+        self._attr_precision = PRECISION_WHOLE
+        if self._client.get_tempscale() == self._client.TSCALE_C:
+            self._attr_temperature_unit = TEMP_CELSIUS
+            self._attr_precision = PRECISION_HALVES
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return the current HVAC mode."""
+        mode = self._client.get_heatmode()
+        return self._balboa_to_ha_heatmode_map[mode]
+
+    @property
+    def hvac_action(self) -> str:
+        """Return the current operation mode."""
+        state = self._client.get_heatstate()
+        if state >= self._client.ON:
+            return CURRENT_HVAC_HEAT
+        return CURRENT_HVAC_IDLE
+
+    @property
+    def fan_mode(self) -> str:
+        """Return the current fan mode."""
+        fanmode = self._client.get_blower()
+        return self._balboa_to_ha_blower_map.get(fanmode, FAN_OFF)
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._client.get_curtemp()
+
+    @property
+    def target_temperature(self):
+        """Return the target temperature we try to reach."""
+        return self._client.get_settemp()
+
+    @property
+    def preset_mode(self):
+        """Return current preset mode."""
+        return self._client.get_heatmode(True)
+
+    async def async_set_temperature(self, **kwargs):
+        """Set a new target temperature."""
+        scale = self._client.get_tempscale()
+        newtemp = kwargs[ATTR_TEMPERATURE]
+        if newtemp > self._client.tmax[self._client.TEMPRANGE_LOW][scale]:
+            await self._client.change_temprange(self._client.TEMPRANGE_HIGH)
+            await asyncio.sleep(1)
+        if newtemp < self._client.tmin[self._client.TEMPRANGE_HIGH][scale]:
+            await self._client.change_temprange(self._client.TEMPRANGE_LOW)
+            await asyncio.sleep(1)
+        await self._client.send_temp_change(newtemp)
+
```

Comment:
In a future PR, we should move this logic into the library.  OK to keep it here for now

Reply:
Acknowledged